CDP4Reporting/Utilities/ProcessedValueSetGenerator.cs

```diff
@@ -0,0 +1,285 @@
+        }
+
+        /// <summary>
+        /// Process the <see cref="ParameterValueSet"/> .
+        /// </summary>
+        /// <param name="parameterSubscriptionValueSet">
+        /// The <see cref="ParameterSubscriptionValueSet"/> to be processed.
+        /// </param>
+        /// <param name="parameterSubscription">
+        /// The <see cref="ParameterSubscription"/> the <see cref="ParameterSubscriptionValueSet"/ belongs to.
+        /// </param>
+        /// <param name="componentIndex">
+        /// The index of the <see cref="ParameterTypeComponent"/>.
+        /// </param>
+        /// <param name="computedValue">
+        /// The manual value of the <see cref="ParameterValueSet"/>.
+        /// </param>
+        /// <param name="processedValueSets">
+        /// A <see cref="Dictionary{Guid,ProcessedValueSet}"/> of ProcessedValueSe that capture the updated <see cref="Thing"/>s with its value validation result
+        /// </param>
+        /// <param name="provider">
+        /// The <see cref="IFormatProvider"/> used to validate.
+        /// </param>
+        /// <param name="validationErrorText">
+        /// <see cref="String"/> that holds the validation error text.
+        /// </param>
+        /// <returns>
+        /// The added or changed <see cref="ProcessedValueSet"/>
+        /// </returns>
+        private ProcessedValueSet ProcessValueSet(ParameterSubscriptionValueSet parameterSubscriptionValueSet, ParameterSubscription parameterSubscription, int componentIndex, string computedValue, ref Dictionary<Guid, ProcessedValueSet> processedValueSets, IFormatProvider provider, out string validationErrorText)
+        {
+            var validationResult = ValidationResultKind.InConclusive;
+
+            var switchKind = parameterSubscriptionValueSet.ValueSwitch;
+            var measurementScale = parameterSubscription.Scale;
+            var parameterType = parameterSubscription.ParameterType;
+
+            validationErrorText = string.Empty;
+
+            if (parameterType != null)
+            {
+                computedValue = computedValue.ToValueSetObject(parameterType).ToValueSetString(parameterType);
+
+                var validManualValue = parameterType.Validate(computedValue, measurementScale, provider);
+
+                if (validManualValue.ResultKind > validationResult)
+                {
+                    validationResult = validManualValue.ResultKind;
+                    validationErrorText = validManualValue.Message;
+                }
+            }
+
+            var valueSetExists = processedValueSets.TryGetValue(parameterSubscriptionValueSet.Iid, out var processedValueSet);
+
+            if (!valueSetExists)
+            {
+                processedValueSet = new ProcessedValueSet(parameterSubscriptionValueSet, validationResult);
+            }
+
+            if (processedValueSet.IsDirty(componentIndex, parameterType, switchKind, computedValue, null, null, null, out var valueSetValues))
```

Comment:
All the "line too long" comments in this file make me wonder if perhaps some of the methods used should be refactored. For example, the `IsDirty` method has 8 parameters.

Reply:
Code was copied from another location. Would take too much time to refactor both locations.